ffs-shared/rule/src/commonMain/kotlin/rule/RuleEval.kt

```diff
@@ -571,19 +547,142 @@ private sealed class RuleExpr<out T> {
         }
 
         @Suppress("PARAMETER_NAME_CHANGED_ON_OVERRIDE")
-        private fun wrapRangeInCollection(range: LongRange): Collection<Long> {
-            if (range.first > range.last) {
+        private fun <T : Comparable<T>> wrapRangeInCollection(
+            range: ClosedRange<T>
+        ): Collection<T> {
+            if (range.start > range.endInclusive) {
                 throw IllegalArgumentException("can't use inverted range")
             }
 
-            return object : ClosedRange<Long> by range, Collection<Long> {
-                override val size: Int get() = (endInclusive - start).toInt()
-                override fun containsAll(elements: Collection<Long>) = elements.all {
+            return object : ClosedRange<T> by range, Collection<T> {
+                override fun containsAll(elements: Collection<T>) = elements.all {
                     range.contains(it)
                 }
-                override fun contains(value: Long) = range.contains(value)
-                override fun iterator() = range.iterator()
+                override fun contains(value: T) = range.contains(value)
+                override val size: Int get() {
+                    TODO("Doesn't need this function")
```

Comment:
Let's use `IllegalAccessException` if we expect this not to be needed. Same below.